pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,261 @@
+				// remove saved profile on change
+				cleanSavedProtoMessage(savedLocalProfileFile)
+				localProfileServer = getconfigCtx.localProfileServer
+			}
+			if localProfileServer != "" {
+				start := time.Now()
+				iteration++
+				localProfileURL, err := getLocalProfileURL(localProfileServer)
+				if err != nil {
+					log.Errorf("getLocalProfileURL: %s", err)
+				} else {
+					if err := getLocalProfileConfig(localProfileURL, iteration, getconfigCtx); err != nil {
+						log.Errorf("localProfileTimerTask getLocalProfileConfig: %s", err)
+					}
+				}
+				ctx.ps.CheckMaxTimeTopic(wdName, "getLocalProfileConfig", start,
+					warningTime, errorTime)
+				publishZedAgentStatus(getconfigCtx)
+			}
+
+		case <-stillRunning.C:
+		}
+		ctx.ps.StillRunning(wdName, warningTime, errorTime)
+	}
+}
+
+func validateAndSetLocalProfile(localProfileBytes []byte, getconfigCtx *getconfigContext) error {
+	var localProfile = &profile.LocalProfile{}
+	err := proto.Unmarshal(localProfileBytes, localProfile)
+	if err != nil {
+		return fmt.Errorf("validateAndSetLocalProfile: Unmarshalling failed: %v", err)
+	}
+	if localProfile.GetServerToken() != getconfigCtx.profileServerToken {
+		// send something to ledmanager ??
+		return fmt.Errorf("validateAndSetLocalProfile: missamtch ServerToken for loacl profile server")
+	}
+	if getconfigCtx.currentProfile != localProfile.GetLocalProfile() {
+		log.Noticef("validateAndSetLocalProfile: profile changed from %s to %s",
+			getconfigCtx.currentProfile, localProfile.GetLocalProfile())
+		getconfigCtx.currentProfile = localProfile.GetLocalProfile()
+		writeReceivedProtoMessageLocalProfile(localProfileBytes)
+		publishZedAgentStatus(getconfigCtx)
+	}
+	return nil
+}
+
+// read saved local profile in case of particular reboot reason
+// if it is success will not read again on the next call
+func readSavedLocalProfile(getconfigCtx *getconfigContext) error {
+	localProfile, err := readSavedProtoMessage(
+		getconfigCtx.zedagentCtx.globalConfig.GlobalValueInt(types.StaleConfigTime),
+		filepath.Join(checkpointDirname, savedLocalProfileFile), false)
+	if err != nil {
+		return fmt.Errorf("lastlocalprofile: %v", err)
+	}
+	if localProfile != nil {
+		log.Function("Using saved local profile")
+		getconfigCtx.readSavedLocalProfile = true
+		return validateAndSetLocalProfile(localProfile, getconfigCtx)
+	}
+	return nil
+}
+
+//prepareLocalProfileServerMap process configuration of network instances
+//to find match with defined localServerURL
+func prepareLocalProfileServerMap(localServerURL string, getconfigCtx *getconfigContext) (map[string]net.IP, error) {
+	res := make(map[string]net.IP)
+
+	netInstanses := getconfigCtx.pubNetworkInstanceConfig.GetAll()
+	u, err := url.Parse(localServerURL)
+	if err != nil {
+		return nil, fmt.Errorf("checkAndPrepareLocalIP: url.Parse: %s", err)
+	}
+	localProfileServerHostname := u.Hostname()
+	localProfileServerIP := net.ParseIP(localProfileServerHostname)
```

Comment:
Should check if this returns nil e.g., if it is a hostname and not a dotted quad.